segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/cache/CacheStorage.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.cache;
+
+import io.pravega.common.util.BufferView;
+import io.pravega.segmentstore.storage.CacheException;
+import java.util.function.Supplier;
+import lombok.NonNull;
+
+/**
+ * Defines a generic, non-associative, block-based Cache Storage.
+ */
+public interface CacheStorage extends AutoCloseable {
+    /**
+     * Gets a value representing the size of one block. For efficiency purposes, it is highly recommended that all inserted
+     * data align to this value (i.e., have a length that is a multiple of this value).
+     *
+     * @return The block alignment, in bytes.
+     */
+    int getBlockAlignment();
+
+    /**
+     * Gets a value representing the maximum size of an entry that can be stored in this {@link CacheStorage}.
+     *
+     * @return The maximum size, in bytes.
+     */
+    int getMaxEntryLength();
```

Comment:
Internally the metadata for each Cache Block is bounded (64 bits). This metadata needs to contain a number of different things, including how long is the entry beginning (or ending) at that block. Since I only have so many bits available,  I need to upper-bound this.

Additionally, inserting a very large entry into the cache is not every efficient. It will take longer to do and its chances of failing (due to the cache being full) are increased. Default is 256MB, which should be plenty for our needs.